sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyType.cs

```diff
@@ -9,54 +9,32 @@ namespace Azure.Security.KeyVault.Keys
     /// <summary>
     /// Supported JsonWebKey key types (kty)
     /// </summary>
-    public enum KeyType : uint
+    public struct KeyType
     {
-        EllipticCurve = 0x0001,
-        EllipticCurveHsm = 0x0002,
-        Rsa = 0x0004,
-        RsaHsm = 0x0008,
-        Octet = 0x0010,
-        Other = 0x0020,
-    }
-
-    public static class KeyTypeExtensions
-    {
-        public static KeyType ParseFromString(string value)
+        /// <summary>
+        /// Supported JsonWebKey key types (kty) as enum
+        /// </summary>
+        public enum KeyTypeValues : uint
         {
-            switch (value)
-            {
-                case "EC":
-                    return KeyType.EllipticCurve;
-                case "EC-HSM":
-                    return KeyType.EllipticCurveHsm;
-                case "RSA":
-                    return KeyType.Rsa;
-                case "RSA-HSM":
-                    return KeyType.RsaHsm;
-                case "oct":
-                    return KeyType.Octet;
-                default:
-                    return KeyType.Other;
-            }
+            EllipticCurve = 0x0001,
+            EllipticCurveHsm = 0x0002,
+            Rsa = 0x0004,
+            RsaHsm = 0x0008,
+            Octet = 0x0010
+            // When a new entry is added, make sure to expose it as a field below.
```

Comment:
Removed `Other` as that would use a slot in the middle of the enum (if we were to add more enum value later). We also don't need it since any value can now be retrieved.

Reply:
Personally I think you need to have the value other, but it should probably be the default, i.e. value 0.  Also this enum isn't used anywhere I would expect that I could construct this struct with the enum, as well as the struct having a Value property which returns this enum type. 

---
In reply to: [293892927](https://github.com/Azure/azure-sdk-for-net/pull/6594#discussion_r293892927) [](ancestors = 293892927)